src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1194,12 +1194,20 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			if (this.useConnectionString) {
-				const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
-				if (connInfo) {
+				try {
+					const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
+					if (!connInfo) {
+						this._logService.error(`${this._providerName}.buildConnectionInfo returned an undefined value.`)
+						this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"), localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string."),);
```

Comment:
it would be nice if you don't have to call the showDialg twice.

Reply:
or define the resource string as a variable so that we don't have 2 copies of the string.

Reply:
This sounds reasonable, but I'm really new to TS and already realized that I was throwing for control flow initially which is not good. Defining the resource strings would mean allocating them for no reason if there is no error which doesn't sound right either.

Having just one `showDialog` call would need it to be outside of the try/catch block (unless using throw flow as in my previous approach, which made it worse style-wise) but then I'd need to cache the result and errors and define several conditions around the call in i.e. a finally block, also considering that the 3rd parameter defines whether the button will be shown or not.

What would be the best practice here?

Reply:
I think you could simply do `throw new Error();` here

Reply:
The error should have the reason, for this case it would be something like `No connection info returned` I would say

Reply:
Not enough to matter for something like ADS.

Reply:
> The error should have the reason, for this case it would be something like `No connection info returned` I would say

It doesn't really add value for the user here. It's the user facing error message and `No connection info returned` is more confusing compared to `Failed to parse the connection string.`, like "returned" from where? In both cases it's clear that parsing the connection string failed, the reason doesn't really matter unless for diagosing, where more details are needed which is not fixed by different phrasing.

Reply:
Realistically this shouldn't be happening - returning undefined isn't valid per the defined API so if someone does that then that means they're implementing it incorrectly. And the message would still be : 

`Failed to parse the connection string: No connection info returned`. Which is both clear enough as to the issue and also lets us know what exactly happened if a user reports this as a bug. The problem with using the same error message for both the cases where it throws and where it returns undefined is we have no way of knowing whether this was an actual error or something else happened that caused it to (incorrectly) return undefined.

Reply:
sounds good, fixed with a935073:

<img width="644" alt="grafik" src="https://user-images.githubusercontent.com/951587/226758104-50150be6-04e3-4c34-b6cd-5f3942834c79.png">